### PR TITLE
adding azure cli 2.0

### PR DESCRIPTION
**Pull request by lakindu95**

### CODE REVIEW

File: src/services/azure-cli/azure-cli.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  });
+} else {
+  docker = new Docker({
+    socketPath: "//./pipe/docker_engine"
+  });
+}
+
+class AzureCLI extends CloudLocal {
+    start() {
+    let optsc = {
```
Comment (dilantha111) [inline src/services/azure-cli/azure-cli.js]: Suggest to use const

File: src/services/azure-cli/azure-cli.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        'StdinOnce': false,
+        'Env': null,
+        'Cmd': ['bash'],
+        'Dns': ['8.8.8.8', '8.8.4.4'],
+        'Image': 'microsoft/azure-cli',
+        'Volumes': {},
+        'VolumesFrom': []
+      };
+      
+      var previousKey,
```
Comment (dilantha111) [inline src/services/azure-cli/azure-cli.js]: Suggest to use const

File: src/services/azure-cli/azure-cli.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          CTRL_P = '\u0010',
+          CTRL_Q = '\u0011';
+      
+      function handler(err, container) {
+        var attach_opts = {stream: true, stdin: true, stdout: true, stderr: true};
```
Comment (dilantha111) [inline src/services/azure-cli/azure-cli.js]: suggest to use const

File: src/services/azure-cli/azure-cli.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      
+        container.attach(attach_opts, function handler(err, stream) {
+          // Show outputs
```
Comment (dilantha111) [inline src/services/azure-cli/azure-cli.js]: suggest that we can handle the errors before rest of the execution

File: src/services/azure-cli/azure-cli.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+          process.stdin.pipe(stream);
+      
+          process.stdin.on('data', function(key) {
+            // Detects it is detaching a running container
+            if (previousKey === CTRL_P && key === CTRL_Q) exit(stream, isRaw);
+            previousKey = key;
+          });
+      
+          container.start(function(err, data) {
+            resize(container);
```
Comment (dilantha111) [inline src/services/azure-cli/azure-cli.js]: can handle container start errors

File: src/services/azure-cli/azure-cli.js (hunk 1 above)
Comment (lakindu95) [reply src/services/azure-cli/azure-cli.js]: added a const

File: src/services/azure-cli/azure-cli.js (hunk 2 above)
Comment (lakindu95) [reply src/services/azure-cli/azure-cli.js]: added a const

File: src/services/azure-cli/azure-cli.js (hunk 3 above)
Comment (lakindu95) [reply src/services/azure-cli/azure-cli.js]: added a const with error handling

File: src/services/azure-cli/azure-cli.js (hunk 4 above)
Comment (lakindu95) [reply src/services/azure-cli/azure-cli.js]: added an error handler

File: src/services/azure-cli/azure-cli.js (hunk 5 above)
Comment (lakindu95) [reply src/services/azure-cli/azure-cli.js]: added an error handle

**Comment by rajikaimal** [comment]
@dilantha111 any more changes needed?

**Pull request merged**